src/Task.cc

```diff
@@ -2393,11 +2393,15 @@ static void set_up_process(Session& session, const ScopedFd& err_fd) {
     spawned_child_fatal_error(err_fd, "error setting up prctl");
   }
 
-  if (0 > prctl(PR_SET_NO_NEW_PRIVS, 1, 0, 0, 0)) {
-    spawned_child_fatal_error(
-        err_fd,
-        "prctl(NO_NEW_PRIVS) failed, SECCOMP_FILTER is not available: your "
-        "kernel is too old. Use `record -n` to disable the filter.");
+  /* If we're in setuid_sudo mode, we have CAP_SYS_ADMIN, so we don't need to
+     set NO_NEW_PRIVS here in order to install the seccomp filter later */
+  if (!session.is_recording() || !session.as_record()->setuid_sudo_mode()) {
```

Comment:
Yeah, probably fine. The reason I didn't do it is because no_new_privs isn't particularly harmful when we're not interested in running privileged executables, but probably doesn't matter too much either.